reciperadar/api/products.py

```diff
@@ -15,36 +15,18 @@ def stream(items):
         yield f"{line}\n"
 
 
-def _product_map(products):
-    product_map = {}
-    for product, nutrition, count, plural_count in products.all():
-        product.nutrition = nutrition
-        product.count = count or 0
-        product.plural_count = plural_count or 0
-        product_map[product.id] = product
-    return product_map
-
-
-def _calculate_depth(product_map, product):
-    if product.parent_id is None:
-        return 0
-    parent = product_map[product.parent_id]
-    return _calculate_depth(product_map, parent) + 1
-
-
-def _product_stream(product_map):
-    for product in product_map.values():
-        depth = _calculate_depth(product_map, product)
-        is_plural = product.plural_count > product.count - product.plural_count
+def _product_stream(products):
+    for product, name, nutrition, count, plural_count in products.all():
+        plural_count = plural_count or 0
+        is_plural = plural_count > count - plural_count
         result = {
-            "product": product.plural if is_plural else product.singular,
-            "recipe_count": product.count,
+            "product": name.plural if is_plural else name.singular,
+            "recipe_count": count,
             "id": product.id,
             "parent_id": product.parent_id,
-            "depth": depth,
         }
-        if product.nutrition:
-            result["nutrition"] = product.nutrition.to_doc()
+        if nutrition:
+            result["nutrition"] = nutrition.to_doc()
```

Comment:
It would be nice to remove this conditional and move the representation into the `result` dictionary declaration.

My guess is that we don't do that because the knowledge graph relies on the absence/presence of the `nutrition` key.  That's not a good reason, though.